softwareconfiguration/mlplan/sklearn/src/main/java/ai/libs/mlplan/sklearn/builder/MLPlanScikitLearnBuilder.java

```diff
@@ -113,10 +109,10 @@ private void setDeterministicDatasetSplitter(final ISupervisedLearnerEvaluatorFa
 	}
 
 	@Override
-	public MLPlan<ScikitLearnWrapper<IPrediction, IPredictionBatch>> build() {
+	public MLPlan<IScikitLearnWrapper> build() {
 		if (!this.skipSetupCheck) {
-			new PythonRequirementDefinition(PYTHON_MINIMUM_REQUIRED_VERSION_REL, PYTHON_MINIMUM_REQUIRED_VERSION_MAJ, PYTHON_MINIMUM_REQUIRED_VERSION_MIN, ArrayUtils.addAll(PYTHON_REQUIRED_MODULES, this.pythonAdditionalRequiredModules))
-			.check(this.pythonConfig);
+			new PythonRequirementDefinition(AScikitLearnWrapper.PYTHON_MINIMUM_REQUIRED_VERSION_REL, AScikitLearnWrapper.PYTHON_MINIMUM_REQUIRED_VERSION_MAJ,
+					AScikitLearnWrapper.PYTHON_MINIMUM_REQUIRED_VERSION_MIN, ArrayUtils.addAll(PYTHON_REQUIRED_MODULES, this.pythonAdditionalRequiredModules)).check(this.pythonConfig);

```

Comment:
this seems a little inconvenient to me to add required modules here

Reply:
also: What about optional modules? one could for instance consider xgboost to be optional. If it is included then fine, otherwise exclude it from the search space?